integration/src/test/java/edu/unc/lib/boxc/integration/web/access/SearchEndpointIT.java

```diff
@@ -240,4 +241,54 @@ public void testSearchWithADifferentStartIndex() throws Exception {
             assertEquals(3, metadata.size());
         }
     }
+
+    @Test
+    public void testSearchPublicUserCannotSeeStaffOnlyObjects() throws Exception {
+        createDefaultObjects();
+        collectionFactory.createCollection(adminUnit1,
+                Map.of("title", "A first collection", "adminGroup", "adminGroup"));
+
+        var getMethod = new HttpGet(SEARCH_URL);
+
+        try (var resp = httpClient.execute(getMethod)) {
+            var metadata = getMetadataFromResponse(resp);
```

Comment:
For this test, it would be good to verify that the id of the restricted collection doesn't show up in the results. We should have access to the id of the new collection from the object returned by `createCollection`.

You can use the streaming api for the search, something like
`metadata.stream().noneMatch(entry -> collectionId.equals(entry.get("id")))`

For the next test, it would be good to do the reverse, verify that the collection id is present. I know the count indirectly confirms that an object is being added or removed, but it would be good to add a little more certainty.